app/src/main/java/com/android/shelfLife/ui/recipes/generateRecipe/GenerateRecipe.kt

```diff
@@ -0,0 +1,499 @@
+package com.android.shelfLife.ui.recipes.generateRecipe
+
+import android.widget.Toast
+import androidx.compose.animation.core.Spring
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.animation.core.spring
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import com.android.shelfLife.R
+import com.android.shelfLife.model.recipe.Recipe.Companion.MAX_SERVINGS
+import com.android.shelfLife.model.recipe.RecipeType
+import com.android.shelfLife.ui.navigation.NavigationActions
+import com.android.shelfLife.ui.navigation.Screen
+import com.android.shelfLife.ui.newoverview.ListFoodItems
+import com.android.shelfLife.ui.recipes.IndividualRecipe.RecipeContent
+import com.android.shelfLife.ui.utils.CustomButtons
+import com.android.shelfLife.ui.utils.CustomTopAppBar
+import com.android.shelfLife.ui.utils.DropdownFields
+import com.android.shelfLife.viewmodel.overview.OverviewScreenViewModel
+import com.android.shelfLife.viewmodel.recipes.RecipeGenerationViewModel
+import kotlin.math.floor
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun GenerateRecipeScreen(
```

Comment:
For some of the composables (name for example) you could use some of the utils composables (which are also used in add/edit food item screen as well as the manual add recipe screen) to avoid code duplication